File: src/test/java/boat/torrent/MagnetDLTest.java
package boat.torrent;

import java.util.List;

import boat.utilities.HttpHelper;
import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;
import static org.junit.jupiter.api.Assertions.assertEquals;

class MagnetDLTest {

    @Test
    void shouldFindTorrents() {
        // Given
        // When
        assertThat(new MagnetDL(new HttpHelper()).searchTorrents("planet").size()).isPositive();
    }

    @Test
    void shouldFindNoTorrents() {
        // Given
        // When
        List<Torrent> torrentList = new MagnetDL(new HttpHelper()).searchTorrents("ThisshouldntbeafindableStringAtall");
        // Then
        assertEquals(0, torrentList.size());
    }
}

File: src/test/java/boat/torrent/searchEngines/MagnetDLTest.kt
package boat.torrent.searchEngines

import boat.torrent.Torrent
import boat.utilities.HttpHelper
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test

internal class MagnetDLTest {
    @Test
    fun shouldFindTorrents() {
        // Given
        // When
        Assertions.assertDoesNotThrow<List<Torrent>> { MagnetDL(HttpHelper()).searchTorrents("planet") }
    }

    @Test
    fun shouldFindNoTorrents() {
        // Given
        // When
        val torrentList = MagnetDL(HttpHelper()).searchTorrents("ThisshouldntbeafindableStringAtall")
        // Then
        Assertions.assertEquals(0, torrentList.size)
    }
}